### PR TITLE
[TOAZ-177] WSM assigning VMs to the wrong managed identity

### PR DESCRIPTION
**Pull request by rtitle**
Depends on https://github.com/broadinstitute/sam/pull/769

1. Commit https://github.com/DataBiosphere/terra-workspace-manager/pull/737/commits/3de63de6981bd2cdb6bdcb73f86c7aa4a37def52 cleanly reverts this revert: https://github.com/DataBiosphere/terra-workspace-manager/pull/723

2. Commit https://github.com/DataBiosphere/terra-workspace-manager/pull/737/commits/9adfd381faea040cdfa5586d10e9b5783d4798ac updates `AssignManagedIdentityAzureVmStep` to do the VM assignment based on the `privateResourceUser` if present (and not assign the VM to a managed identity otherwise).

### CODE REVIEW

**Comment by ddietterich** [comment]
> Commit https://github.com/DataBiosphere/terra-workspace-manager/commit/2c103bb3e1e0193ab90821c334652ea06a47d555 updates AssignManagedIdentityAzureVmStep to do the VM assignment based on the privateResourceUser if present; or the calling user otherwise.

What is the ramification of using the calling user? Is the VM somehow bound to the calling user after this point? That would worry me a bit. In the case of the Leo application making the call, is there a `privateResourceUser` specified? Do we ever want these VMs without a `privateResourceUser`? Like does that open up a security problem?

**Comment by rtitle** [comment]
> What is the ramification of using the calling user? Is the VM somehow bound to the calling user after this point? That would worry me a bit. In the case of the Leo application making the call, is there a privateResourceUser specified? Do we ever want these VMs without a privateResourceUser? Like does that open up a security problem?

Leo always sets `privateResourceUser` to the user owning the VM, so I'm mainly just concerned with that use case. That's a good point that assigning the VM to the calling user would not be safe if this is a shared resource, or an application resource with no `privateResourceUser`. Maybe it's more correct to _only_ do the VM assignment if `privateResourceUser` is specified. wdyt?

**Comment by rtitle** [comment]
I made the change described and it seems to be working with Leo in my test environment. Requesting some reviews…

File: integration/build.gradle
Comment (rtitle) [inline integration/build.gradle:33]: Note: will update to non-SNAP once https://github.com/broadinstitute/sam/pull/769 is merged

**Comment by ddietterich** [comment]
> Maybe it's more correct to only do the VM assignment if privateResourceUser is specified.
That sounds better to me. If we run into a use case for the no-privateResourceUser, we can sort it out then.

**Review by Qi77Qi — APPROVED** [review]
some nits...lgtm once Sam PR is approved...

**Review by ddietterich — APPROVED** [review]
I made some minor code comments.
Otherwise, LGTM

**Review by zykonda — APPROVED** [review]

File: service/src/main/java/bio/terra/workspace/service/crl/CrlService.java
Comment (rtitle) [reply service/src/main/java/bio/terra/workspace/service/crl/CrlService.java:183]: Yeah +1.. I think there's some other copy-paste duplication in this class w.r.t. Azure clients, will try to clean up

File: integration/build.gradle
Comment (rtitle) [reply integration/build.gradle:75]: I made https://broadworkbench.atlassian.net/browse/TOAZ-181

This also came up in a [Rawls PR](https://github.com/broadinstitute/rawls/pull/1946/files#r932333901), which is now using TCL.

**Pull request merged**
